httpc.go

```diff
@@ -198,6 +200,21 @@ func (r *Request) EncodeXML(v interface{}) *Request {
 	return r
 }
 
+// AuthBasic sets parameters to perform basic authentication
+func (r *Request) AuthBasic(user, password string) *Request {
+	r.httpAuthFunc = func(c *http.Request) {
+		c.SetBasicAuth(user, password)
+	}
+	return r
+}
+
+// AuthDigest sets parameters to perform digest-based authentication
+func (r *Request) AuthDigest(user, password string) *Request {
+	t := dac.NewTransport(user, password)
+	r.client.Transport = &t
```

Comment:
This will conditionally break if you use the client certificate feature as well (see https://github.com/fako1024/httpc/blob/604e6395d6a52b2782eae9ef3db28e5e71b9d057/httpc.go#L131)

Reply:
Very good point. I could add a check to your ClientCertificate methods (since they return an error anyway) to break in case the Transport is not the default one... In addition I could add another check in Run() to validate consistency and fail gracefully...

Reply:
NVM, this is not a good idea, I'll drop the AuthDigest completely. It's such a rare use case and comes with so many pitfalls (there's multiple places where the code rellies on the transport being a http.Transport, it's not worth the effort. I'll just use the ModifyClient() method. Good catch!